File: admin/get cli/e_bandobas/lib/app/jsondata/EventData/Event.dart
class Event {
  Event(
      {this.id,
      this.eventName,
      this.eventDetails,
      this.eventStartDate,
      this.eventEndDate});

  Event.fromJson(dynamic json) {
    id = json['id'];
    eventName = json['eventName'];
    eventDetails = json['eventDetails'];
    eventStartDate = json['eventStartDate'];
    eventEndDate = json['eventEndDate'];
  }
  num? id;
  String? eventName;
  String? eventDetails;
  String? eventStartDate;
  String? eventEndDate;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['eventName'] = eventName;
    map['eventDetails'] = eventDetails;
    map['eventStartDate'] = eventStartDate;
    map['eventEndDate'] = eventEndDate;
    return map;
  }
}

File: gujarat_police_backend/src/main/java/com/shyam/gujarat_police/entities/Event.java
package com.shyam.gujarat_police.entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.Date;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
public class Event extends BaseModel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String eventName;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String eventDetails;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
    private Date eventStartDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
    private Date eventEndDate;

    @JsonIgnore
    @OneToMany(mappedBy = "event", cascade = CascadeType.PERSIST)
    private List<AssignPolice> assignPolice;

    @JsonIgnore
    @OneToMany(mappedBy = "event", cascade = CascadeType.PERSIST)
    private List<EventPoliceCount> eventPoliceCountList;

    @JsonIgnore
    @OneToMany(mappedBy = "event", cascade = CascadeType.PERSIST)
    private List<Police> policeList;



    @Override
    public String toString() {
        return "Event{" +
                "id=" + id +
                ", eventName='" + eventName + '\'' +
                ", eventDetails='" + eventDetails + '\'' +
                ", eventStartDate=" + eventStartDate +
                ", eventEndDate=" + eventEndDate +
                '}';
    }
}
